src/OpenTK.Windowing.Common/Input/WindowIcon.cs

```diff
@@ -7,97 +7,30 @@
 // of the MIT license. See the LICENSE file for details.
 //
 
-using System;
-using System.Runtime.InteropServices;
-
 namespace OpenToolkit.Windowing.Common.Input
 {
     /// <summary>
-    /// Stores a window icon. A window icon is defined
-    /// as a 2-dimensional buffer of RGBA values.
+    /// Stores an array of <see cref="Image" />s, meant for use as window icons.
     /// </summary>
     public class WindowIcon
     {
-        /// \internal
         /// <summary>
         /// Initializes a new instance of the <see cref="WindowIcon" /> class.
         /// </summary>
-        protected internal WindowIcon()
-        {
-        }
+        private WindowIcon() { }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="WindowIcon" /> class.
         /// </summary>
-        /// <param name="width">The icon pixel width.</param>
-        /// <param name="height">The icon pixel height.</param>
-        private WindowIcon(int width, int height)
+        /// <param name="images">An array of <see cref="Image" />s, which will be used as the window icons.</param>
+        public WindowIcon(Image[] images)
```

Comment:
I think a params array would be nice and would make it easier to use without any disadvantage.

Reply:
What do you mean by "params array"?

Reply:
I think he means using the params modifier. So it becomes `public WindowIcon(params Image[] images`

Reply:
I'd never even heard of that before! Cool beans, added in latest commit.